File: Excel/650 Challenge.py
import pandas as pd
path = "650 Top 3 Across Columns.xlsx"
input = pd.read_excel(path, usecols="A:E", nrows=15)
test = pd.read_excel(path, usecols="G", nrows=5).values.flatten().tolist()

result = input.apply(lambda col: ', '.join(map(str, sorted(col.dropna().astype(int).unique())[:3]))).tolist()

print(result == test) # True

File: Excel/600-699/650/650 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/650 Top 3 Across Columns.xlsx"
input = read_excel(path, range = "A1:E15")
test  = read_excel(path, range = "G1:G6") %>% pull()

result = input %>%
  summarise(across(everything(), ~ paste(sort(unique(na.omit(.)))[1:3], collapse = ", "))) %>%
  as.list() %>%
  unlist()

all.equal(result, test)
# [1] TRUE